### PR TITLE
Task/use amqp durable option

### PR DESCRIPTION
**Pull request by AlvaroVega**
https://www.rabbitmq.com/mqtt.html#durability
Durable = true is QoS level 1

### CODE REVIEW

File: lib/constants.js [hunk 1]
```diff
@@ -56,6 +56,7 @@ module.exports = {
 
     AMQP_DEFAULT_EXCHANGE: 'amq.topic',
     AMQP_DEFAULT_QUEUE: 'iotaqueue',
+    AMQP_DEFAULT_DURABLE: true,
```
Comment (fgalan) [inline lib/constants.js]: I understand that the config.amqp.durable (= AMQP_DEFAULT_DURABLE env var) was already included in config.js and documentation but up to now the code didn't use it. Is my understanding correct?

Comment (AlvaroVega) [reply lib/constants.js]: Yes, that is

File: lib/bindings/AMQPBinding.js [hunk 2]
```diff
@@ -86,6 +86,14 @@ function start(callback) {
         queue = constants.AMQP_DEFAULT_QUEUE;
     }
 
+    var durable;
+
+    if (config.getConfig() && config.getConfig().amqp && config.getConfig().amqp.durable) {
+        durable = config.getConfig().amqp.durable;
+    } else {
+        durable = constants.AMQP_DEFAULT_DURABLE;
+    }
```
Comment (fgalan) [inline lib/bindings/AMQPBinding.js]: Looking how durable is being processed here I guess it should be (inside config.amqp):

```
    durable: true,
```

However, in config.js I see (https://github.com/telefonicaid/iotagent-json/blob/master/config.js#L100):

```
    options: { durable: true }
```

Maybe config.js should be fixed?

Comment (AlvaroVega) [reply lib/bindings/AMQPBinding.js]: fixed in https://github.com/telefonicaid/iotagent-json/pull/448/commits/26774c007854239ca7bdb953dace103c10c4c714

File: lib/constants.js (hunk 1 above)
Comment (fgalan) [reply lib/constants.js]: NTC

**Pull request merged**